website/content/en/docs/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,396 @@
+
+```sh
+$ mkdir $HOME/projects/memcached-operator
+$ cd $HOME/projects/memcached-operator
+$ kubebuilder init --domain example.com --license apache2 --owner "The Memcached Operator authors" --repo github.com/example-inc/memcached-operator
+$ cd memcached-operator
+```
+
+Make sure to activate the go module support by running `$ export GO111MODULE=on`.
+
+## Create an API
+
+Create a new API and a new kind (CRD) for the operator.
+
+In case of memcached operator, a new API named `cache/v1aplha1` (group/version) and a new kind `Memcached` is created using the following command.
+
+`kubebuilder create api --group cache --version v1alpha1 --kind Memcached`
+
+Press `y` when asked for creating resource and controller. This will scaffold the project and create the files `api/<version>/<kind>_types.go` and `controller/<kind>_types.go`.
+
+For memcached operator project, `api/v1alpha1/memcached_types.go` and `controller/memcached_controller.go` are generated.
+
+```sh
+Create Resource [y/n]
+y
+Create Controller [y/n]
+y
+Writing scaffold for you to edit...
+api/v1alpha1/memcached_types.go
+controllers/memcached_controller.go
+Running make:
+$ make
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/admin/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+go fmt ./...
+go vet ./...
+```
+
+### Define Spec and Status
+
+Copy over the API spec and status from the SDK project's `pkg/apis/<group>/<version>/<kind>_types.go` to Kubebuilder's `api/<version>/<kind>_types.go`.
+
+In our example, `MemcachedSpec` and `MemcachedStatus` from [`pkg/apis/cache/v1aplha1/memcached_types.go`][memcached_types] are copied to `api/v1alpha1/memcached_types.go`.
+
+```go
+type MemcachedSpec struct {
+	// Specify size, the number of pods
+	Size int32 `json:"size"`
+}
+
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+```
+
+### CRD markers
+
+Include any of the CRD generation or validation markers defined for your API type and fields in `api/<version>/<kind>_types.go`.
+
+For `memcached-operator` the following markers present in `pkg/apis/cache/v1aplha1/memcached_types.go` are copied to `api/v1alpha1/memcached_types.go`.
+
+Example:
+```Go
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=memcacheds,scope=Namespaced
+```
+
+### Controller logic
+
+Copy over the `Reconcile()` code and the controller logic from your existing controller at `pkg/controller/<kind>/<kind>_controller.go` to the new controller's reconcile function in Kubebuilder project at `controllers/<kind>_controller.go`.
+
+In our example [`pkg/controller/memcached/memcached_controller.go`][memcached_controller] is copied to `controllers/memcached_controller.go`.
+
+Particularly, note the following changes:
+
+1. Update the import paths and names as necessary:
```

Comment:
If we're just copying over the Reconcile() function code, do we need to update the default imports scaffolded by kubebuilder? Those should already be correct right?
https://github.com/operator-framework/operator-sdk/blob/master/example/kb-memcached-operator/memcached_controller.go.tmpl#L17-L32

So perhaps no need for this step.

Reply:
So maybe we just have the steps be:
1) copying over the reconcile function 
2) updating the naming conventions and import aliases in the reconciler code 
3) Copy over any remaining controller pkgs and helpers from `pkg/controller/<kind>` to `controllers/`